speed_time.py

```diff
@@ -0,0 +1,47 @@
+import matplotlib.pyplot as plt
+from datasets import load_dataset
+import pandas as pd
+
+def plot_speed(df):
+  """Plots of speed vs time.
+   Time_interval will be computed as df['end']-df['start']
+
+   speed, the note per second can be computed as 1/time_interval
+
+   incase the time exceed 120 second we will convert it into minutes
+
+  Args:
+    df: A Pandas DataFrame containing the notes from a MIDI file.
+
+  Returns:
+    A  plot of matplotlib .
+  """
+
+# Calculate the time intervals and speed
+  time_intervals = df["end"] - df["start"]
+  speed = 1 / time_intervals
```

Comment:
I think this is not the speed we're looking for. We are interested with speed defined as "number of notes per minute", and what you are calculating looks like `1/note_duration` for each note.